WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormTableViewDataSource.swift

```diff
@@ -231,4 +240,11 @@ private extension ProductFormTableViewDataSource {
         }
         cell.updateUI(switchableViewModel: switchableViewModel)
     }
+
+    func configureWarningRow(cell: UITableViewCell, viewModel: ProductFormSection.SettingsRow.WarningViewModel) {
+        guard let cell = cell as? ImageAndTitleAndTextTableViewCell else {
+            fatalError()
```

Comment:
I think it's worth crashing here because this specific case is when an unexpected cell type is registered for the row, which should be tested whenever we add a row and the crash for devs could be helpful for locating the error. A message would certainly be helpful, added here: https://github.com/woocommerce/woocommerce-ios/pull/2627/commits/26278eca6fb15617ed89658f6fcd5aafba0f75ec

Lemme know if you think if it's better not crashing for the user though! In some more unpredictable cases, we use `assertionFailure`+`DDLogError` to only crash in development and log to application log when the user submits a ticket. I think for very clear dev error cases, it's more clear and easier crashing for all since it's easy to test such cases and it's logged to Sentry. But I'm also open to other approaches!

Reply:
That's good enough for me! 

>  I think for very clear dev error cases, it's more clear and easier crashing for all since it's easy to test such cases and it's logged to Sentry.

I agree. Sometimes a `fatalError()` doesn't relay that. So the message that you added helps. 🙂